### ISSUE TITLE
Vault using outdated OCI library which blocks automatic unseal in new regions

### ISSUE THREAD

**Issue by sosiskin2000** [hashicorp/vault#17147]
**Describe the bug**
There is a new Regions and Realms which available only in V65 versions of https://github.com/oracle/oci-go-sdk. All examples describes that we have to use this version for module import github.com/oracle/oci-go-sdk/v65/* but in code you have github.com/oracle/oci-go-sdk/* and version when i'm using (go get: added [github.com/oracle/oci-go-sdk](http://github.com/oracle/oci-go-sdk) v24.3.0+incompatible), which is outdated. .
New regions and realms as example provided only in https://github.com/oracle/oci-go-sdk/blob/master/common/regions.go#L202

**To Reproduce**
1. Tenancy in OC9 new OCI realm
2. Use instruction to setup unseal according to site - https://www.vaultproject.io/docs/configuration/seal/ocikms
3. Vault will use crypto endpoint with oraclecloud.com in  oc1 region. But must use oraclecloud9.com.

**Expected behavior**
This documentation working https://www.vaultproject.io/docs/configuration/seal/ocikms in oc9 new realm.

**Environment:**
* Vault Server Version 1.11
* Server Operating System/Architecture: Oracle Linux/Centos

**Additional context**
Its a blocker for whole Oracle Org using Hashicorp.

**Comment by mpalmi** [comment]
Looks like this was also opened here: https://github.com/hashicorp/vault-plugin-auth-oci/issues/17

Since the [docs](https://developer.hashicorp.com/vault/docs/auth/oci) recommend opening and tracking issues in that repo, I'm going to go ahead and close this issue out.

**Comment by sosiskin2000** [comment]
@mpalmi but there is 2 different part of code. 
1) In vault and unseal - https://www.vaultproject.io/docs/configuration/seal/ocikms
2) In https://github.com/hashicorp/vault-plugin-auth-oci/issues/17 in Auth plugin.
And this code using different imports.

**Comment by mpalmi** [comment]
Good point @sosiskin2000. I'll go ahead and re-open this until we're sure everything behaves as expected. Thanks again for the report!

**Comment by heatherezell** [comment]
Thanks for the report! This issue was filed against Vault `v1.11`, which is no longer a supported release.

We're unable to investigate bugs against unsupported versions. If you're still experiencing this on a current supported release of Vault, please open a new issue with updated environment details and we'll take a look.

For a list of currently supported versions, see the [Vault version support page](https://github.com/hashicorp/vault/blob/main/.release/versions.hcl). For Enterprise long-term support information, see the [LTS tracker](https://developer.hashicorp.com/vault/docs/updates/lts-tracker).
